signal/signal.go

```diff
@@ -34,20 +34,24 @@ type Interceptor struct {
 
 	// quit is closed when instructing the main interrupt handler to exit.
 	quit chan struct{}
+
+	// exitCode is the exit code status when the main interrupt handler exits
+	exitCode chan int
```

Comment:
Worth adding a comment here that we **must** set a value here, otherwise the `GetExitCode` function will block.